packages/big-design/src/components/Input/Input.tsx

```diff
@@ -22,134 +22,94 @@ interface PrivateProps {
   forwardedRef: Ref<HTMLInputElement>;
 }
 
-interface InputState {
-  focus: boolean;
-}
-
 export type InputProps = Props & React.InputHTMLAttributes<HTMLInputElement>;
 
-class StyleableInput extends React.PureComponent<InputProps & PrivateProps, InputState> {
-  static Description = FormControlDescription;
-  static Error = FormControlError;
-  static Label = FormControlLabel;
-
-  readonly state: InputState = {
-    focus: false,
-  };
-
-  private readonly uniqueId = uniqueId('input_');
-
-  render() {
-    const { chips, description, disabled, error, label, labelId, forwardedRef, onChipDelete, ...props } = this.props;
-    const id = this.getId();
-
-    return (
-      <div>
-        {this.renderLabel()}
-        {this.renderDescription()}
-        <StyledInputWrapper disabled={disabled} error={error} focus={this.state.focus}>
-          {this.renderIconLeft()}
-          <StyledInputContent chips={chips}>
-            {this.renderChips()}
-            <StyledInput
-              {...props}
-              disabled={disabled}
-              chips={chips}
-              error={error}
-              id={id}
-              onBlur={this.onInputBlur}
-              onFocus={this.onInputFocus}
-              ref={forwardedRef}
-            />
-          </StyledInputContent>
-
-          {this.renderIconRight()}
-        </StyledInputWrapper>
-      </div>
-    );
-  }
-
-  private getId() {
-    const { id } = this.props;
-
-    return id ? id : this.uniqueId;
-  }
-
-  private onInputFocus = (event: React.FocusEvent<HTMLInputElement>) => {
-    const { onFocus } = this.props;
-
-    this.setFocus(true);
+const StyleableInput: React.FC<InputProps & PrivateProps> = ({
+  chips,
+  description,
+  disabled,
+  error,
+  forwardedRef,
+  label,
+  labelId,
+  onChipDelete,
+  ...props
+}) => {
+  const [focus, setFocus] = useState(false);
+  const id = useMemo(() => (props.id ? props.id : uniqueId('input_')), [props.id]);
+
+  const onInputFocus = (event: React.FocusEvent<HTMLInputElement>) => {
```

Comment:
🍸 What do you think if we rename this event handler method to`handleInputFocus` or just `handleFocus`?